src/storage/schema.rs

```diff
@@ -0,0 +1,265 @@
+use std::fmt;
+
+use crate::errors::*;
+use crate::storage::Transaction;
+
+pub const LATEST_SCHEMA_VERSION: SchemaVersion = SchemaVersion {
+    major: 0,
+    minor: 7,
+    patch: 0,
+    revision: 1,
+};
+
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct SchemaVersion {
+    major: u32,
+    minor: u32,
+    patch: u32,
+    revision: u32,
+}
+
+impl SchemaVersion {
+    pub fn from_tuple(major: u32, minor: u32, patch: u32, revision: u32) -> Self {
+        Self {
+            major,
+            minor,
+            patch,
+            revision,
+        }
+    }
+}
+
+impl fmt::Display for SchemaVersion {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "{}.{}.{}-{}",
+            self.major, self.minor, self.patch, self.revision
+        )
+    }
+}
+
+pub fn current_schema_version(tx: &mut Transaction) -> Result<Option<SchemaVersion>> {
```

Comment:
While the Go code returns a default version, I found it more idiomatic in rust to make it optional. A default version is still used in the `upgrade` function.